cimbar/cimbar.py

```diff
@@ -62,23 +65,141 @@ def bits_per_op():
     return conf.BITS_PER_SYMBOL + BITS_PER_COLOR
 
 
+def use_split_mode():
+    return getattr(conf, 'SPLIT_MODE', True)
+
+
 def num_cells():
     return conf.CELL_DIM_Y*conf.CELL_DIM_X - (conf.MARKER_SIZE_X*conf.MARKER_SIZE_Y * 4)
 
 
+def num_fountain_blocks():
+    return bits_per_op() * 2
+
+
 def capacity(bits_per_op=bits_per_op()):
     return num_cells() * bits_per_op // 8;
 
 
 def _fountain_chunk_size(ecc=conf.ECC, bits_per_op=bits_per_op(), fountain_blocks=conf.FOUNTAIN_BLOCKS):
+    fountain_blocks = fountain_blocks or num_fountain_blocks()
     return capacity(bits_per_op) * (conf.ECC_BLOCK_SIZE-ecc) // conf.ECC_BLOCK_SIZE // fountain_blocks
 
 
+def _get_expected_fountain_headers(headers, bits_per_symbol=conf.BITS_PER_SYMBOL, bits_per_color=BITS_PER_COLOR):
+    import bitstring
+    from bitstring import Bits, BitStream
+
+    # it'd be nice to use the frame id as well, but sometimes we skip frames.
```

Comment:
Worth noting: I'm currently ignoring this advice in `libcimbar` and using the frame_id anyway. It's only going to mess up one decode, what's the harm? :grimacing: 

(TODO: do the math in libcimbar to not mess up that one frame)